pandas/io/common.py

```diff
@@ -276,7 +276,27 @@ def ZipFile(*args, **kwargs):
     ZipFile = zipfile.ZipFile
 
 
-def _get_handle(path, mode, encoding=None, compression=None):
+def _get_inferred_compression(filepath_or_buffer, compression):
+    if compression == 'infer':
+        if isinstance(filepath_or_buffer, compat.string_types):
+            if filepath_or_buffer.endswith('.gz'):
+                inferred_compression = 'gzip'
+            elif filepath_or_buffer.endswith('.bz2'):
+                inferred_compression = 'bz2'
+            elif filepath_or_buffer.endswith('.zip'):
+                inferred_compression = 'zip'
+            elif filepath_or_buffer.endswith('.xz'):
+                inferred_compression = 'xz'
+            else:
+                inferred_compression = None
+        else:
+            inferred_compression = None
+    else:
+        inferred_compression = compression
+    return inferred_compression
+
+
+def _get_handle(path, mode, encoding=None, compression=None, is_txt=True):
```

Comment:
hmm, ok, let's call this ``is_text=True`` then

and pls add a doc-string to ``_get_handle`` (should have been there, but since now adding params)......

thanks